### PR TITLE
Increase error message truncation limit

### LINKED ISSUE #13046: Error message truncation may be too strict
**Describe the bug**

In #11460 we've implemented a truncation on error messages. This was done to prevent error messages from exceeding the maximum message size. Currently the messages are limited to 500 characters.

We've had a support case asking us where the remainder of the error message has gone. 500 characters may be too strict. A regular stack trace will quickly exceed this limit.

Generally users would have logging on their workers which should be able to show them the full message. Obviously this isn't visible in Operate and is an extra step for users. If we could provide a better UX by increasing this limit we should consider this.

**To Reproduce**

Throw an error on a Job with > 500 characters.

**Expected behavior**

I can see more than just the first 500 characters of my error.

-------

https://jira.camunda.com/browse/SUPPORT-17090

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

The purpose of this PR is to update error message size from 500 to 10k chars for a better user experience. This PR will help user to find easily the error in a process instance failure.

## Related issues

closes #13046 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by remcowesterhoud** [comment]
@nicpuppa please add a short summary under "description" explaining what this PR fixes 🙂

File: engine/src/test/java/io/camunda/zeebe/engine/processing/job/JobThrowErrorTest.java [hunk 1]
```diff
@@ -407,7 +407,33 @@ public void shouldTruncateErrorMessage() {
...
+  public void shouldNotTruncateErrorMessage() {
+    // given
+    final var job = ENGINE.createJob(jobType, PROCESS_ID);
+    final String errorMessage = "*".repeat(DEFAULT_MAX_ERROR_MESSAGE_SIZE);
+
+    // when
+    final Record<JobRecordValue> failedRecord =
+        ENGINE.job().withKey(job.getKey()).withErrorMessage(errorMessage).throwError();
+
+    final Record<IncidentRecordValue> incident =
```
Comment (remcowesterhoud) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/job/JobThrowErrorTest.java:430]: 🙃 Very nitpicky, don't feel pressured to change this if you don't want to, but we could use `var` to make it fit on a single line.

File: engine/src/test/java/io/camunda/zeebe/engine/processing/job/FailJobTest.java [hunk 2]
```diff
@@ -336,7 +336,28 @@ public void shouldTruncateErrorMessage() {
...
+  public void shouldNotTruncateErrorMessage() {
+    // given
+    final Record<JobRecordValue> job = ENGINE.createJob(jobType, PROCESS_ID);
+    final String errorMessage = "*".repeat(DEFAULT_MAX_ERROR_MESSAGE_SIZE);
+
+    // when
+    final Record<JobRecordValue> failedRecord =
+        ENGINE.job().withKey(job.getKey()).withErrorMessage(errorMessage).fail();
+
+    final Record<IncidentRecordValue> incident =
```
Comment (remcowesterhoud) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/job/FailJobTest.java:353]: 🙃 Very nitpicky, don't feel pressured to change this if you don't want to, but we could use `var` to make it fit on a single line.

**Review by remcowesterhoud — APPROVED** [review]
Great work @nicpuppa 🚀 The only suggestions I have are very nitpicky, so feel free to ignore them.

I see that the backport of https://github.com/camunda/zeebe/pull/12066 failed and I don't see any motivation why this hasn't been done manually. I feel like we should backport this. Would you be willing to try and backport both to `stable/8.0` and `stable/8.1`. And then of course also backport this one to `stable/8.2` as you've already marked?

File: engine/src/test/java/io/camunda/zeebe/engine/processing/job/JobThrowErrorTest.java (hunk 1 above)
Comment (nicpuppa) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/job/JobThrowErrorTest.java]: Yes, make sense. Thank you

File: engine/src/test/java/io/camunda/zeebe/engine/processing/job/FailJobTest.java (hunk 2 above)
Comment (nicpuppa) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/job/FailJobTest.java]: Yes, make sense. Thank you

**Comment by nicpuppa** [comment]
> Great work @nicpuppa 🚀 The only suggestions I have are very nitpicky, so feel free to ignore them.
> 
> I see that the backport of #12066 failed and I don't see any motivation why this hasn't been done manually. I feel like we should backport this. Would you be willing to try and backport both to `stable/8.0` and `stable/8.1`. And then of course also backport this one to `stable/8.2` as you've already marked?

Thank you @remcowesterhoud for the review! Can you help me with the backport ?

**Pull request merged**